control-plane/pkg/util/lock_map.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
```

Comment:
`util`? A better name like `lockmap`?

Reply:
IMO it would be nice if we had a `util` package and stash stuff in it. But, I am ok with your suggestion

Reply:
Let me rename it, put it on a different package and also make the value generic. But in a separate PR.